docs/development/testguide.rst

```diff
@@ -665,6 +665,32 @@ a real-world example::
    so the `astropy.tests.helper.catch_warnings` context manager is
    preferred.
 
+Testing exceptions
+==================
+
+Just like the handling of warnings described above, tests that are
+designed to trigger certain errors should verify that an exception of
+the expected type is raised in the expeced place.  This is efficiently
+done by running the tested code inside the
+`pytests.raises <https://docs.pytest.org/en/latest/assert.html#assertraises>`_
```

Comment:
please remove the hard wired link, "`pytest.raises`" should be resolved via intersphinx anyway.